unittests/AST/StructuralEquivalenceTest.cpp

```diff
@@ -838,6 +838,33 @@ TEST_F(StructuralEquivalenceLambdaTest, LambdaClassesWithEqFields) {
   EXPECT_TRUE(testStructuralMatch(L0, L1));
 }
 
+TEST_F(StructuralEquivalenceLambdaTest,
+       LambdaClassesWithDifferentSpellingLocations) {
+  auto t = makeDecls<LambdaExpr>("void f() { auto L = [](int){}; }",
+                                 "void f() { auto L =  [](int){}; }",
+                                 Lang_CXX11, lambdaExpr(), lambdaExpr());
+  CXXRecordDecl *L0 = get<0>(t)->getLambdaClass();
+  CXXRecordDecl *L1 = get<1>(t)->getLambdaClass();
+  EXPECT_FALSE(testStructuralMatch(L0, L1));
+}
+
+TEST_F(StructuralEquivalenceLambdaTest,
+       LambdaClassesWithDifferentExpansionLocations) {
+  auto t = makeDecls<LambdaExpr>(
+      R"(
+      #define LAMBDA [](){}
+      void f() { auto L = LAMBDA; }
+      )",
+      R"(
+      #define LAMBDA [](){}
+      void f() { auto L =       LAMBDA; }
+      )",
+      Lang_CXX11, lambdaExpr(), lambdaExpr());
+  CXXRecordDecl *L0 = get<0>(t)->getLambdaClass();
+  CXXRecordDecl *L1 = get<1>(t)->getLambdaClass();
+  EXPECT_FALSE(testStructuralMatch(L0, L1));
+}
+
```

Comment:
These last 2 tests look a bit strange, these could be equivalent too. Why is this needed? Probably it is OK to make lambdas never structurally equivalent?

Reply:
According to the standard, every new lambda expression introduces a new type.
So inside one TU, every different lambda should have a different type, yes. However, across TUs, if the lambda is coming from a header with the very same location then I think we should not create a new type, because that is essentially the very same lambda. We can satisfy both if we check for the location, can't we?

Reply:
Normally it should work. If an imported TU uses a copy of a header (with exact same contents or re-formatted) that is used by the already imported things it may not work. And theoretically two code parts where only the amount of whitespace is different should be equivalent.
Probably it is possible to use a similar check like at unnamed structs: Check the position of the lambda Decl in its parent (not sure if it is good always)?